app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/LoginActivity.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * LoginActivity.java
+ * Version: 1.0
+ * Date: October 18, 2020
+ * Copyright (c) 2020. Book Friends Team
+ * All rights reserved.
+ * github URL: https://github.com/CMPUT301F20T21/Book_Friends
+ */
+
+package com.cmput301f20t21.bookfriends.ui.login;
+
+import android.os.Bundle;
+import android.text.Editable;
+import android.view.View;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+/**
+ * Activity that will display the login page and handle user inputs
+ */
+public class LoginActivity extends AppCompatActivity {
+    private TextInputLayout usernameLayout;
+    private TextInputLayout passwordLayout;
+    private TextInputEditText usernameField;
+    private TextInputEditText passwordField;
+
+    /**
+     * android lifecycle method, grab the Layout and EditText fields
+     * as well as adding a simple textChangedListener
+     * @param savedInstanceState - the saved objects, should contain nothing for this activity
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        usernameLayout = findViewById(R.id.login_username_layout);
+        usernameField = findViewById(R.id.login_username_field);
```

Comment:
I believe the field is coupled with the layout right? I think you could do usernameField = usernameLayout.getEditText()

Reply:
getEditText() does not work for this, material design text field use TextInputEditText class instead of EditText class

Reply:
I could do EditText usernameField = usernameLayout.getEditText() with no error locally.

Reply:
fixed, need a explicit cast to make it work